gradle-palantir-java-format/src/main/groovy/com/palantir/javaformat/gradle/ConfigureJavaFormatterXml.groovy

```diff
@@ -49,19 +49,18 @@ class ConfigureJavaFormatterXml {
     }
 
     private static void configureOnSaveAction(Node onSaveOptions) {
-        def myRunOnSave = matchOrCreateChild(onSaveOptions, 'option', [name: 'myRunOnSave'])
-
-        def myRunOnSaveAlreadySet = Optional.ofNullable(myRunOnSave.attribute('value'))
-                .map(Boolean::parseBoolean)
-                .orElse(false)
-
-        myRunOnSave.attributes().put('value', 'true')
+        // If myRunOnSave is set to true, IntelliJ removes it. If it's set to false, we still need it to run.
+        // So we should just remove it so we do run on save.
+        matchChild(onSaveOptions, 'option', [name: 'myRunOnSave']).ifPresent { myRunOnSave ->
+            onSaveOptions.remove(myRunOnSave)
+        }
 
         def myAllFileTypesSelectedAlreadySet = matchChild(onSaveOptions, 'option', [name: 'myAllFileTypesSelected'])
                 .map { Boolean.parseBoolean(it.attribute('value')) }
-                .orElse(true)
```

Comment:
This is the crux of the bug. If `myAllFileTypesSelected` did not already exist, we would think it is already set to true and make no further changes, despite the default being false.